deepchem/models/layers.py

```diff
@@ -471,18 +471,83 @@ def call(self, inputs):
 
 
 def _cosine_dist(x, y):
-  """Computes the inner product (cosine distance) between two tensors.
+  """Computes the inner product (cosine similarity) between two tensors.
+  
+  This assumes that the two input tensors contain rows of vectors where 
+  each column represents a different feature. The output tensor will have
+  elements that represent the inner product between pairs of normalized vectors
+  in the rows of x and y. The two tensors need to have the same number of columns,
+  because one cannot take the dot product between vectors of different lengths.
+  For example, in sentence similarity and sentence classification tasks,
+  the number of columns is the embedding size. In these tasks, the rows of the
+  input tensors would be different test vectors or sentences. The input tensors
+  themselves could be different batches. Using vectors or tensors of all 0s
+  should be avoided.
+
+  Method
+  ------
+  The vectors in the input tensors are first l2-normalized such that each vector
+  has length or magnitude of 1. The inner product (dot product) is then taken 
+  between corresponding pairs of row vectors in the input tensors and returned.
+
+  Examples
+  --------
+  The cosine similarity between two equivalent vectors will be 1. The cosine
+  similarity between two equivalent tensors (input tensors where the elements are
+  the same), will be a tensor of ones. In this scenario, if the input tensors
+  a and y were each of shape (n,p), where each element in x and y were the same, then 
+  the output tensor would be a tensor of shape (n,n) with 1 in every entry.
+  The cosine similarity between two orthogonal vectors will be 0 (by definition).
+  If every row in x is orthogonal to every row in y, then the output will be a tensor
+  of 0s.
+
+  ```python
```

Comment:
Could you swap this over to doctest style? So this would look like

```
>>> import tensorflow as tf
>>> import deepchem.models.layers as layers
...
```
That way, the CI can run the code in the docstrings to make sure it remains correct